tests/integration_tests.rs

```diff
@@ -0,0 +1,844 @@
+        .unwrap();
+
+    submit_result.assert_success();
+    // List IDs start from 0 and increment
+    let list_id: u64 = 0;
+
+    // Approve the list
+    let total_amount = NearToken::from_yoctonear(250_000_000_000_000_000_000_000_000); // 250 NEAR
+    
+    // Get contract balance before payouts
+    let contract_balance_before = near_api::Account(contract_id.clone())
+        .view()
+        .fetch_from(&network_config)
+        .await
+        .unwrap()
+        .data
+        .amount;
+
+    near_api::Contract(contract_id.clone())
+        .call_function("approve_list", json!({ "list_ref": list_id }))
+        .unwrap()
+        .transaction()
+        .deposit(total_amount)
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Process first batch (100 payments)
+    near_api::Contract(contract_id.clone())
+        .call_function(
+            "payout_batch",
+            json!({
+                "list_ref": list_id,
+                "max_payments": 100
+            }),
+        )
+        .unwrap()
+        .transaction()
+        .gas(near_sdk::Gas::from_tgas(300))
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Process second batch (100 payments)
+    near_api::Contract(contract_id.clone())
+        .call_function(
+            "payout_batch",
+            json!({
+                "list_ref": list_id,
+                "max_payments": 100
+            }),
+        )
+        .unwrap()
+        .transaction()
+        .gas(near_sdk::Gas::from_tgas(300))
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Process third batch (50 payments)
+    near_api::Contract(contract_id.clone())
+        .call_function(
+            "payout_batch",
+            json!({
+                "list_ref": list_id,
+                "max_payments": 100
+            }),
+        )
+        .unwrap()
+        .transaction()
+        .gas(near_sdk::Gas::from_tgas(300))
+        .with_signer(user_id.clone(), user_signer.clone())
+        .send_to(&network_config)
+        .await
+        .unwrap()
+        .assert_success();
+
+    // Verify all recipients received their payments
+    for (i, recipient) in recipients.iter().enumerate() {
+        let balance = near_api::Account(recipient.clone())
+            .view()
+            .fetch_from(&network_config)
+            .await
+            .unwrap()
+            .data
+            .amount;
+        
+        // Each recipient started with 1 NEAR and should have received 1 NEAR
+        // So balance should be approximately 2 NEAR (minus some gas fees)
+        assert!(
+            balance.as_yoctonear() > 1_500_000_000_000_000_000_000_000, // At least 1.5 NEAR
```

Comment:
You should be able to assert the exact amount ( 2 NEAR ), since gas is not deducted from the transfer amount.